app/routes/circulars/circulars.lib.ts

```diff
@@ -76,6 +76,28 @@ export function formatDateISO(date: number) {
   return new Date(date).toISOString().replace(/\.\d+Z$/, 'Z')
 }
 
+/** convert a ms value to mm-dd-yyy format */
+export function formatDateMMDDYYYY(date: number) {
+  const dateObj = new Date(date)
+  var dateStr = dateObj.toLocaleDateString('en-US', {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+  })
+  return dateStr.replace(/\//g, '-') // replace / with -
+}
+/** convert a ms value to YYYY-MM-DD format */
+export function formatDateYYYYMMDD(date: number) {
+  const dateObj = new Date(date)
+  var dateStr = dateObj.toISOString().split('T')[0]
+  return dateStr
+}
+
+/** convert a date in format mm-dd-yyyy (or YYYY-MM_DD) to ms since 01/01/1970 */
+export function formatDateEpoch(date: string) {
+  return new Date(date).getTime()
+}
```

Comment:
Take care: when you search for records with dates between `yyyy-mm-dd` (lower limit) and `YYYY-MM-DD` (upper limit) that you are searching between the times `yyyy-mm-ddT00:00:00Z` and `YYYY-MM-DDT23:59:59Z` (so that hits include the whole day that comprises the upper end of the interval).

Reply:
I made a small change in 55e52d9 so the endDate, when defined, should include circulars recorded on the date of the endDate. I did this by just adding the number of milliseconds in a day to the endTime constant, which is in unix format,  when it's defined inside of the search function. Wasn't sure if that's sufficient or if it would it be better to look for a more elegant implementation.